File: sdks/dotnet/src/Resources/ReturnsResource.cs
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class ReturnsResource
    {
        private readonly PuxbayClient _client;
        public ReturnsResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<ReturnRequest>> ListAsync(int page = 1)
        {
            return await _client.GetAsync<PaginatedResponse<ReturnRequest>>($"returns/?page={page}");
        }

        public async Task<ReturnRequest> GetAsync(string returnId)
        {
            return await _client.GetAsync<ReturnRequest>($"returns/{returnId}/");
        }

        public async Task<ReturnRequest> CreateAsync(ReturnRequest returnReq)
        {
            return await _client.PostAsync<ReturnRequest>("returns/", returnReq);
        }

        public async Task<ReturnRequest> ApproveAsync(string returnId)
        {
            return await _client.PostAsync<ReturnRequest>($"returns/{returnId}/approve/", null);
        }
    }
}

File: sdks/php/src/Resources/ReturnsResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\ReturnRequest;
use Puxbay\Models\PaginatedResponse;

class ReturnsResource extends BaseResource
{
    public function list(int $page = 1): PaginatedResponse
    {
        $data = $this->get("returns/?page={$page}");
        return PaginatedResponse::fromArray($data, fn($item) => ReturnRequest::fromArray($item));
    }

    public function getReturn(string $returnId): ReturnRequest
    {
        $data = $this->get("returns/{$returnId}/");
        return ReturnRequest::fromArray($data);
    }

    public function create(ReturnRequest $returnData): ReturnRequest
    {
        $data = $this->post('returns/', $returnData->toArray());
        return ReturnRequest::fromArray($data);
    }

    public function approve(string $returnId): ReturnRequest
    {
        $data = $this->post("returns/{$returnId}/approve/");
        return ReturnRequest::fromArray($data);
    }
}
